ui/src/main/webapp/src/app/core/loading-indicator.service.ts

```diff
@@ -0,0 +1,90 @@
+import {Injectable} from "@angular/core";
+import {SlimLoadingBarService} from 'ng2-slim-loading-bar';
+import {EventBusService} from "./events/event-bus.service";
+import {LoadingSomethingFinishedEvent, LoadingSomethingStartedEvent} from "./events/windup-event";
+import {Observable} from "rxjs/Observable";
+
+/**
+ * Uses the ng2-slim-loading-bar component to indicate loading progress.
+ * See https://www.npmjs.com/package/ng2-slim-loading-bar
+ *
+ * In our HTTP service wrapper (handling OAuth), we fire an event, which is then caught by our `LoadingIndicatorService`,
+ * which wraps the `SlimLoaderBarService`, and keeps track of how many HTTP requests there are in progress.
+ * Then it counts the percent and puts it to a scale of 20 to 90 %.
+ * When all are done, it stays at 90 % for about a second, and then `complete()` is called.
+ */
+@Injectable()
+export class LoadingIndicatorService {
+
+    constructor(
+        private _slimBarService: SlimLoadingBarService,
+        private _eventBusService: EventBusService,
+    ) {
+        // Register the LoadingSomething event listeners.
+        this._eventBusService.onEvent
+            .filter(event => event.isTypeOf(LoadingSomethingStartedEvent))
+            .subscribe((event: LoadingSomethingStartedEvent) => this.loadingStarted(event) )
+        this._eventBusService.onEvent
+            .filter(event => event.isTypeOf(LoadingSomethingFinishedEvent))
+            .subscribe((event: LoadingSomethingFinishedEvent) => this.loadingFinished(event) )
+    }
+
+    public getSlimService(){
+        return this._slimBarService;
+    }
+
+
+    private counter: number = 0;
+    private max: number = void 0;
+    private hadError: boolean = false;
+
+    private reset() {
+        this.counter = 0;
+        this.max = void 0;
+        //console.log(`reset(), counter ${this.counter}`);
+    }
+
+    public loadingStarted(event: LoadingSomethingStartedEvent){
+        this.counter++;
+        this.max = this.counter;
+        //console.log(`event received START, counter ${this.counter}, max ${this.max}`);
+        this.updateProgress();
+    }
+
+    public loadingFinished(event: LoadingSomethingFinishedEvent){
+        this.counter--;
+        // On error, change the bar color to red.
+        event.getResponse().filter(res => res.status < 200 && res.status >= 400).subscribe(_ => { this.hadError = true; });
+        //console.log(`event received FINISH, counter ${this.counter}, max ${this.max}`);
+        this.updateProgress();
+    }
+
+    private updateProgress() {
+        //console.log(`updateProgress(), counter ${this.counter}, max ${this.max}`);
+        if (this.counter == 0) {
+            //console.log("INDI All finished, setting to complete.");
+            this._slimBarService.height = "2px";
```

Comment:
However consider that it's the way the component is designed by the author. See their readme.